src/NServiceBus.Persistence.DynamoDB/SynchronizedStorage/StorageSession.cs

```diff
@@ -74,7 +74,7 @@ public async Task Commit(CancellationToken cancellationToken = default)
 
         public void Dispose()
         {
-
+            disposed = true;
```

Comment:
Disposing multiple times should be allowed due to the contract. This would try to release locks multiple times because we are not clearing the cleanup actions. Should we guard for that or at least clear the cleanup actions at the end of release so that release becomes NoOp?